include/swift/AST/DiagnosticEngine.h

```diff
@@ -724,6 +725,11 @@ namespace swift {
     /// diagnostic message.
     std::unique_ptr<diag::LocalizationProducer> localization;
 
+    /// This allocator will retain localized diagnostic strings for the duration
+    /// of compiler invocation.
+    llvm::BumpPtrAllocator localizationAllocator;
+    llvm::Optional<llvm::StringSaver> localizationSaver;
```

Comment:
This doesn't need to be optional since it's always initialized in the constructor